test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1115,6 +1128,22 @@ private void wipeDataDirectories() {
         }
     }
 
+    private void markNodeDataDirsAsPendingForWipe(Node node) {
+        assert Thread.holdsLock(this);
+        NodeEnvironment nodeEnv = getInstanceFromNode(NodeEnvironment.class, node);
```

Comment:
while we are at it maybe add a Node.getNodeEnvironment() method that doesn't go through guice?